doofinder-for-woocommerce/includes/class-landing.php

```diff
@@ -0,0 +1,438 @@
+            if($hashid_by_lang === $hashid) {
+                $desired_lang = $language['code'];
+                break;
+            }
+        }
+
+        return $desired_lang;
+    }
+
+    /**
+     * Format the URL by appending the slug to the home URL while considering different URL formats.
+     *
+     * @param string $home_url The home URL for the desired language.
+     * @param string $slug The slug to be appended to the URL.
+     *
+     * @return string The formatted URL with the slug included.
+     */
+    private function formated_url($home_url, $slug) {
+        $slug = "df/{$slug}";
+        if(strpos($home_url, '?lang=') !== false) {
+            $formated_url = str_replace('?lang=', "{$slug}/?lang=", $home_url);
+        } else {
+            $formated_url = "{$home_url}{$slug}";
+        }
+        return $formated_url;
+    }
+
+    /**
+     * Redirect to a given formatted URL.
+     *
+     * @param string $formated_url The URL to which the redirection will occur.
+     */
+    private function redirect($formated_url) {
+        header("Location: $formated_url");
+        exit;
+    }
+
+    /**
+     * Retrieves landing page information based on the provided hashid and slug.
+     *
+     * @param string $hashid The unique identifier (hashid).
+     * @param string $slug The slug of the landing page.
+     *
+     * @return array An array containing landing page information or an error message if the page is not well-constructed.
+     */
+    public function get_landing_info($hashid, $slug) {
+        // Determine the hashid to use based on the provided parameter or settings
+        $hashid = !empty($hashid) ? $hashid : Settings::get_search_engine_hash($this->current_language);
+
+        // Define an error message for cases where required parameters are missing
+        $error_not_set = ['error' => "The page is not well constructed. The hashid or slug is missing."];
+
+        // Check if the necessary parameters are available or return the error message
+        $this->landing_data = self::have_params($hashid, $slug) ? $this->landing_api->get_landing_info($hashid, $slug) : $error_not_set;
+
+        // If landing page data is available, build its blocks
+        if (isset($this->landing_data['data']))
+            $this->build_blocks($hashid);
+
+        // Return the landing page information or error message
+        return $this->landing_data;
+    }
+
+
+    /**
+     * Generates the HTML content for a landing page based on the provided data and landing slug.
+     *
+     * @param array $landing_data  An array containing landing page information or an error message if the page is not well-constructed.
+     * @param string $landing_slug The slug of the landing page.
+     *
+     * @return string The HTML content for the landing page.
+     */
+    public function get_landing_html($landing_data, $landing_slug)
+    {
+        ob_start();
+    ?>
```

Comment:
Edu por qué esto de cerrar aquí el PHP en vz de hacer

$html = <<<EOD
Example of string
spanning multiple lines
using heredoc syntax.
EOD;

o algo así? no se cortar ahí el PHP me parece raro

Reply:
igual es que es la forma de hacerlo con wordpress y ya está, sólo pregunto por curiosidad

Reply:
Justamente iba a comentar eso. En WordPress he visto esta forma de hacerlo bastantes veces, así que debe ser como un estándar.

Reply:
Pues básicamente lo he hecho así por costumbre, es una forma de hacerlo en wordpress. Entiendo que tampoco habría problema en usar Heredoc, Le voy a dar una vuelta para hacerlo de esa manera si lo veis más conveniente!